File: Data Structures & Algorithms/valid-binary-search-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isValidBST(TreeNode root) {
        return helper(root, -1001, 1001);
    }
    public boolean helper(TreeNode root, int l, int r) {
        if(root==null) return true;
        if(root.val >= r || root.val <= l)
            return false;
        return (helper(root.left, l, root.val) && helper(root.right, root.val, r));
    }
}

File: Data Structures & Algorithms/house-robber-iii/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rob(self, root: Optional[TreeNode]) -> int:
        def helper(node: Optional[TreeNode]) -> List[int]:
            if not node:
                return [0,0]
            a = helper(node.left)
            b = helper(node.right)
            r = node.val + a[1]+b[1]
            nr = max(a[0],a[1]) + max(b[0],b[1])
            return [r,nr]
            
        ans = helper(root)
        return max(ans[0],ans[1])

        [2,3], [3,0]